scilpy/tracking/tools.py

```diff
@@ -1,7 +1,139 @@
 # -*- coding: utf-8 -*-
+import logging
 
+import nibabel as nib
 import numpy as np
 
+from dipy.core.sphere import HemiSphere
+from dipy.data import get_sphere
+from dipy.direction import (DeterministicMaximumDirectionGetter,
+                            ProbabilisticDirectionGetter,
+                            PTTDirectionGetter)
+from dipy.direction.peaks import PeaksAndMetrics
+
+from scilpy.reconst.utils import (find_order_from_nb_coeff,
+                                  get_b_matrix, get_maximas)
+
+
+def get_direction_getter(
+    in_odf, algo, sphere, sub_sphere, theta, sh_basis,
+    voxel_size, sf_threshold
+):
+    """ Return the direction getter object.
+
+    Parameters
+    ----------
+    in_odf: str
+        Path to the input odf file.
+    algo: str
+        Algorithm to use for tracking. Can be 'det', 'prob', 'ptt' or 'eudx'.
+    sphere: str
+        Name of the sphere to use for tracking.
+    sub_sphere: int
+        Number of subdivisions to use for the sphere.
+    theta: float
+        Angle threshold for tracking.
+    sh_basis: str
+        Name of the sh basis to use for tracking.
+    voxel_size: float
+        Voxel size of the input data.
+    sf_threshold: float
+        Spherical function-amplitude threshold for tracking.
+
+    Return
+    ------
+    dg: dipy.direction.DirectionGetter
+        The direction getter object.
+    """
+
+    odf_data = nib.load(in_odf).get_fdata(dtype=np.float32)
+
+    sphere = HemiSphere.from_sphere(
+        get_sphere(sphere)).subdivide(sub_sphere)
+
+    # Theta depends on user choice and algorithm
+    theta = get_theta(theta, algo)
+
+    # Heuristic to find out if the input are peaks or fodf
+    non_zeros_count = np.count_nonzero(np.sum(odf_data, axis=-1))
+    non_first_val_count = np.count_nonzero(np.argmax(odf_data, axis=-1))
+
+    if algo in ['det', 'prob', 'ptt']:
+        if non_first_val_count / non_zeros_count > 0.5:
+            logging.warning('Input detected as peaks. Input should be'
+                            'fodf for det/prob, verify input just in case.')
```

Comment:
Shouldn't we ask user if input is fODF or peak, rather than try to discover ourselves?